src/code/z_sram.c

```diff
@@ -224,9 +226,21 @@ static Inventory sDebugSaveInventory = {
         ITEM_SLINGSHOT, ITEM_OCARINA_FAIRY, ITEM_BOMBCHU,      ITEM_HOOKSHOT,    ITEM_ARROW_ICE,   ITEM_FARORES_WIND,
         ITEM_BOOMERANG, ITEM_LENS,          ITEM_BEAN,         ITEM_HAMMER,      ITEM_ARROW_LIGHT, ITEM_NAYRUS_LOVE,
         ITEM_BOTTLE,    ITEM_POTION_RED,    ITEM_POTION_GREEN, ITEM_POTION_BLUE, ITEM_POCKET_EGG,  ITEM_WEIRD_EGG,
-    },                                                              // items
-    { 50, 50, 10, 30, 1, 1, 30, 1, 50, 1, 1, 1, 1, 1, 1, 1 },       // ammo
-    0x7777,                                                         // equipment
+    },                                                        // items
+    { 50, 50, 10, 30, 1, 1, 30, 1, 50, 1, 1, 1, 1, 1, 1, 1 }, // ammo
+    // equipment
+    (((1 << EQUIP_INV_SWORD_KOKIRI) << (EQUIP_TYPE_SWORD * 4)) |
+     ((1 << EQUIP_INV_SWORD_MASTER) << (EQUIP_TYPE_SWORD * 4)) |
+     ((1 << EQUIP_INV_SWORD_BGS) << (EQUIP_TYPE_SWORD * 4))) |
+        (((1 << EQUIP_INV_SHIELD_DEKU) << (EQUIP_TYPE_SHIELD * 4)) |
+         ((1 << EQUIP_INV_SHIELD_HYLIAN) << (EQUIP_TYPE_SHIELD * 4)) |
+         ((1 << EQUIP_INV_SHIELD_MIRROR) << (EQUIP_TYPE_SHIELD * 4))) |
+        (((1 << EQUIP_INV_TUNIC_KOKIRI) << (EQUIP_TYPE_TUNIC * 4)) |
+         ((1 << EQUIP_INV_TUNIC_GORON) << (EQUIP_TYPE_TUNIC * 4)) |
+         ((1 << EQUIP_INV_TUNIC_ZORA) << (EQUIP_TYPE_TUNIC * 4))) |
+        (((1 << EQUIP_INV_BOOTS_KOKIRI) << (EQUIP_TYPE_BOOTS * 4)) |
+         ((1 << EQUIP_INV_BOOTS_IRON) << (EQUIP_TYPE_BOOTS * 4)) |
+         ((1 << EQUIP_INV_BOOTS_HOVER) << (EQUIP_TYPE_BOOTS * 4))),
```

Comment:
This at least looks like it could use further macros. There are other places, of course, but this is the hardest to read.

Reply:
ikr (also the poor formatting isn't helping)

Maybe `#define EQUIP_INV_MASK(equipType, equipInv) ((1 << (equipInv)) << ((equipType) * 4))`? I'm not sure it would make stuff much more readable tbh but at least it abstracts away the bitfield usage

Reply:
Yeah, that would be the most general way, I suppose; the alternative would be make one that's specifically the debug equips with a comment that it's everything. Maybe do the ones elsewhere that use `+ () * 4`, as well.

Reply:
This is actually the only place where that macro can be used... looks like this
```c
    // equipment
    EQUIP_INV_MASK(EQUIP_TYPE_SWORD, EQUIP_INV_SWORD_KOKIRI) |
        EQUIP_INV_MASK(EQUIP_TYPE_SWORD, EQUIP_INV_SWORD_MASTER) |
        EQUIP_INV_MASK(EQUIP_TYPE_SWORD, EQUIP_INV_SWORD_BGS) |
        EQUIP_INV_MASK(EQUIP_TYPE_SHIELD, EQUIP_INV_SHIELD_DEKU) |
        EQUIP_INV_MASK(EQUIP_TYPE_SHIELD, EQUIP_INV_SHIELD_HYLIAN) |
        EQUIP_INV_MASK(EQUIP_TYPE_SHIELD, EQUIP_INV_SHIELD_MIRROR) |
        EQUIP_INV_MASK(EQUIP_TYPE_TUNIC, EQUIP_INV_TUNIC_KOKIRI) |
        EQUIP_INV_MASK(EQUIP_TYPE_TUNIC, EQUIP_INV_TUNIC_GORON) |
        EQUIP_INV_MASK(EQUIP_TYPE_TUNIC, EQUIP_INV_TUNIC_ZORA) |
        EQUIP_INV_MASK(EQUIP_TYPE_BOOTS, EQUIP_INV_BOOTS_KOKIRI) |
        EQUIP_INV_MASK(EQUIP_TYPE_BOOTS, EQUIP_INV_BOOTS_IRON) |
        EQUIP_INV_MASK(EQUIP_TYPE_BOOTS, EQUIP_INV_BOOTS_HOVER),
```
I don't think it's enough of an improvement to justify having a macro just for that one use

Reply:
I improved the formatting with extra parentheses, and suggested another way to write this below (without a macro)